src/main/kotlin/com/github/g3force/oc/OcExecTask.kt

```diff
@@ -5,57 +5,51 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.TaskAction
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
 import java.io.BufferedReader
 import java.io.File
 import java.io.InputStreamReader
 
 abstract class OcExecTask : DefaultTask() {
-
-    private val logger: Logger = LoggerFactory.getLogger(OcExecTask::class.java)
-
-    @Input
-    var args: List<String> = emptyList()
-
     @Input
-    var showOutput: Boolean = false
+    var args: List<String> = ArrayList()
 
     @TaskAction
     open fun doAction() {
-        execute(args)
+        execute(emptyList())
     }
 
     fun execute(arguments: List<String>) {
-        val command: MutableList<String> = mutableListOf(findOcBinary())
+        val config = project.extensions.getByType(OcPluginExtension::class.java)
+        val binary = findOcBinary(config)
+
+        val command: MutableList<String> = mutableListOf(binary)
         command.addAll(arguments)
+        command.addAll(args)
+
+        if (config.insecure) {
+            command.add("--insecure-skip-tls-verify")
+        }
 
         logger.info("Executing: ${command.joinToString(" ")}")
 
-        val pb = ProcessBuilder(command)
+        val processBuilder = ProcessBuilder(command)
                 .directory(File("."))
                 .redirectErrorStream(true)
-        pb.environment()["KUBECONFIG"] = "build/kube.config"
-        val proc = pb.start()
-        val errorCode = proc.waitFor()
+        processBuilder.environment()["KUBECONFIG"] = "build/kube.config"
+        val process = processBuilder.start()
+        val errorCode = process.waitFor()
 
-        val reader = BufferedReader(InputStreamReader(proc.inputStream))
+        val reader = BufferedReader(InputStreamReader(process.inputStream))
         val output = IOUtils.readLines(reader).joinToString("\n")
 
-        logger.info("Output: \n${output}")
-
-        if (showOutput) {
-            println(command.joinToString(" "))
-            println(output)
-        }
+        logger.debug("Output: \n${output}")
 
         if (errorCode != 0) {
-            throw GradleException("Non zero error code '$errorCode' for command ${pb.command()}:\n${output}")
+            throw GradleException("Non zero error code '$errorCode' for command ${processBuilder.command()}:\n${output}")
         }
     }
 
-    private fun findOcBinary(): String {
-        val config = project.extensions.getByType(OcPluginExtension::class.java)
+    private fun findOcBinary(config: OcPluginExtension): String {
```

Comment:
I did not pass the full config here by intention, but retrieving the config again is also not ideal. What about just passing `config.ocBinary` instead?

Reply:
For my local fork, I've made a downloader for client binary with fallback to this property. Wanted to publish as a separate PR. Will change to `config.ocBinary` for now

Reply:
Looking forward to that separat PR :) Would be really helpful.